src/jit/ee_il_dll.cpp

```diff
@@ -439,15 +439,45 @@ unsigned           Compiler::eeGetArgSize(CORINFO_ARG_LIST_HANDLE list, CORINFO_
         assert(argTypeJit != CORINFO_TYPE_REFANY || structSize == 2*sizeof(void*));
 
 #if FEATURE_MULTIREG_ARGS
-#ifdef _TARGET_ARM64_
+        // For each target that supports passing struct args in multiple registers 
+        // apply the target specific rules for them here:
+#if defined(_TARGET_ARM64_)
+        // For ARM64 structs that are larger that 32 bytes are
+        // passed by reference to a local copy of the value type
```

Comment:
In general I try to avoid calling the VM to determine if a struct is an HFA since it might be an expensive operation.
If the struct is larger than 32 bytes it can not be an HFA so it will be passed by reference.